libs/src/api/serviceProvider.ts

```diff
@@ -122,34 +130,53 @@ class ServiceProvider extends Base {
    * @param {any[]} discoveryNodes the verbose list of discovery nodes to select from
    * @param {(node: { delegateOwnerWallet: string }) => boolean} filter an optional filter step to remove certain nodes
    */
-  async getUniquelyOwnedDiscoveryNodes ({ quorumSize, discoveryNodes = [], filter = (node) => true, useWhitelist = true }) {
+  async getUniquelyOwnedDiscoveryNodes({
+    quorumSize,
+    discoveryNodes = [],
+    filter = (_) => true,
+    useWhitelist = true
+  }: {
+    quorumSize: number
+    discoveryNodes?: ServiceWithEndpoint[]
+    filter: (node: ServiceWithEndpoint) => boolean
+    useWhitelist?: boolean
+  }) {
     if (!discoveryNodes || discoveryNodes.length === 0) {
       // Whitelist logic: if useWhitelist is false, pass in null to override internal whitelist logic; if true, pass in undefined
       // so service selector uses internal whitelist
-      discoveryNodes = await this.discoveryProvider.serviceSelector.findAll({ verbose: true, whitelist: useWhitelist ? undefined : null })
+      discoveryNodes = (await this.discoveryProvider.serviceSelector.findAll({
+        verbose: true,
+        whitelist: useWhitelist ? undefined : null
+      })) as ServiceWithEndpoint[]
```

Comment:
asumming the discoveryProvider here is not typed?

Reply:
There are some typing issues that go all the way down to `allRequests` - sometimes it returns Service and sometimes ServiceWithEndpoint. Gonna avoid poking this bees nest for now